tests/acceptance/bootstrap/SharingNgContext.php

```diff
@@ -1800,24 +1779,46 @@ public function checkIfShareExists(string $share, string $sharee, string $sharer
 		$sharedWithMeList = HttpRequestHelper::getJsonDecodedResponseBodyContent($response)->value;
 		$foundShareInSharedWithMe = false;
 		foreach ($sharedWithMeList as $item) {
-			if ($item->name === $share && $item->createdBy->user->displayName === $this->featureContext->getDisplayNameForUser($sharer)) {
-				$foundShareInSharedWithMe = true;
+			if ($item->name === $share) {
+				foreach ($item->remoteItem->permissions as $permission) {
+					$shareCreator = $permission->invitation->invitedBy->user->displayName;
+					if ($shareCreator === $this->featureContext->getDisplayNameForUser($sharer)) {
+						$foundShareInSharedWithMe = true;
+						break;
+					}
+				}
+				break;
 			}
 		}
-		Assert::assertTrue($foundShareInSharedWithMe, "Share '$share' was not found in the shared-with-me list");
+		Assert::assertSame($shouldExist, $foundShareInSharedWithMe, "Share '$share' was not found in the shared-with-me list");
 	}
 
 	/**
-	 * @Then user :sharee should have a share :share shared by user :sharer
+	 * @Then user :sharee should have a share :share shared by user :sharer from space :space
```

Comment:
Is it possible to merge the step def as `(should|should not) have a share...` ?

Reply:
should be